### PR TITLE
Once we decide to force-evict a page, do it directly

### PR DESCRIPTION
**Pull request by michaelcahill**
(rather than setting read_gen and hoping page release agrees).

### CODE REVIEW

File: src/include/btree.i [hunk 1]
```diff
@@ -945,12 +915,14 @@ __wt_page_release_busy(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
 	 */
 	locked = WT_ATOMIC_CAS4(ref->state, WT_REF_MEM, WT_REF_LOCKED);
 	WT_TRET(__wt_hazard_clear(session, page));
-	if (!locked)
+	if (!locked) {
+		WT_TRET(EBUSY);
```
Comment (agorrod) [inline src/include/btree.i]: This is the only part of the change that I'm suspicious about. Could we be returning an EBUSY to the direct caller of ```__wt_page_release_evict``` when we returned success before?

Comment (michaelcahill) [inline src/include/btree.i]: That was an intentional change, but not based on any testing.  I thought we should sleep if the page is already locked by another thread...  I'll back it out and see if it helps with the column store truncate hangs.

File: src/btree/bt_cursor.c [hunk 2]
```diff
@@ -948,7 +948,7 @@ __cursor_truncate(WT_SESSION_IMPL *session,
 	} else {
 		do {
 			WT_RET(__wt_btcur_remove(start));
-			for (;;) {
+			for (ret = 0;;) {
```
Comment (michaelcahill) [inline src/btree/bt_cursor.c]: This is subtle: without it, we ended up at the end of the scan (`start` == `stop`) and `ret == WT_RESTART`, and got stuck in an infinite loop.  Only with column stores, and I'm not sure exactly where the `WT_RESTART` was coming from, but it's clearly a weakness of these loops.

**Comment by michaelcahill** [comment]
@agorrod, this is now passing tests, can you please review for real?

**Comment by agorrod** [comment]
lgtm, I'll merge.

**Pull request merged**